ast_tools/pattern.py

```diff
@@ -0,0 +1,109 @@
+import re
+import ast
+import itertools
+
+# Globals dictionary to evaluate type names in node patterns
+eval_globals = {}
+exec('from ast import *', eval_globals)
```

Comment:
I suspect there's a way to do this without exec (introspecting the AST module and populating the dictionary), but this seems like it should work fine.  Although I suspect @cdonovick would object on principle ;)

Reply:
Yeah that's fine. I changed it to use `getattr(ast, name)` which should be good enough for now until we need greater flexibility.